### PR TITLE
[Telemetry] Add the correct StartupType values for the startup event

### PR DESCRIPTION
**Pull request by iainx**
fixes VSTS #678313

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/options/MonoDevelopProperties.xml [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <Properties>
     <Property key="MonoDevelop.Core.FirstRun" value="True" />
     <Property key="MonoDevelop.Core.Gui.Dialog.TipOfTheDayView.ShowTipsAtStartup" value="False" />
+	<Property key="MonoDevelop.Core.RestartRequested" value="False" />
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/options/MonoDevelopProperties.xml]: Spacing is off

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs [hunk 2]
```diff
@@ -842,13 +848,33 @@ static IdeCustomizer LoadBrandingCustomizer ()
...
 		{
 			var assetType = StartupAssetType.FromStartupInfo (startupInfo);
+			StartupType startupType = StartupType.Normal;
+
+			if (startupInfo.Restarted && !IdeApp.IsInitialRunAfterUpgrade) {
+				startupType = StartupType.ConfigurationChange; // Assume a restart without upgrading was the result of a config change
+			} else if (IdeApp.IsInitialRun) {
+				startupType = StartupType.FirstLaunch;
+			} else if (Debugger.IsAttached) {
+				startupType = StartupType.LaunchedAsDebugger;
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs]: I don't think LaunchedAsDebugger applies to VSMac. On Windows this happens when an application crashes and the debug button is clicked on the crash dialog. This is unrelated to the VSMac process being debugged or not.

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs]: We could be debugging a bundle via `Debug Application`.

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs]: might be worth taking these out as they skew results

Comment (iainx) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs]: I think I meant `DebuggerPresent` not `LaunchedAsDebugger`

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs [hunk 3]
```diff
@@ -842,13 +848,33 @@ static IdeCustomizer LoadBrandingCustomizer ()
...
+		}
+
 		static StartupMetadata GetStartupMetadata (StartupInfo startupInfo, IPlatformTelemetryDetails platformDetails, Dictionary<string, long> timings)
 		{
 			var assetType = StartupAssetType.FromStartupInfo (startupInfo);
+			StartupType startupType = StartupType.Normal;
+
+			if (startupInfo.Restarted && !IdeApp.IsInitialRunAfterUpgrade) {
+				startupType = StartupType.ConfigurationChange; // Assume a restart without upgrading was the result of a config change
+			} else if (IdeApp.IsInitialRun) {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs]: Should we consider doing:
```
if (IdeApp.IsInitialRunAfterUpgrade) {
    startupType = StartupType.FirstLaunch;
} else if (IdeApp.IsInitialRun) {
    startupType = StartupType.FirstLaunchSetup;
}
```

Comment (iainx) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs]: No, `FirstLaunch` is only ever sent once, on the initial launch

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs]: We pay the cost of creating the MEF cache after upgrade, so we need to notify that (otherwise, we'll just have a trend of startup being higher on release and slowly normalizing the the older version.

**Comment by iainx** [comment]
@monojenkins backport release-7.6

**Comment by iainx** [comment]
@monojenkins backport release-7.7
